### PR TITLE
RFE - support template, severity and facility options

### PR DESCRIPTION
**Pull request by nhosoi**
RFE - support template, severity and facility options
- template options to the files & formats output
- severity and facility options to the files input

(issue #271)

Playbook example
```
- name: "Ensure that the role runs with parameters from files input to forwards outputs"
  hosts: all
  tasks:
    - name: "Ensure that the role runs with parameters
      from files input to forwards outputs"
      vars:
        logging_outputs:
          - name: forwardToLogServer
            type: forwards
            target: psyslog.example.com
            udp_port: 514
            template: RSYSLOG_SyslogProtocol23Format
        logging_inputs:
          - name: "php:"
            type: files
            input_log_path: /var/log/php_errors.log
            severity: err
        logging_flows:
          - name: flow
            inputs:
              - "php:"
            outputs:
              - forwardToLogServer
      include_role:
        name: linux-system-roles.logging
```
Generated rsyslog config file example
```
==> 90-input-files-php:.conf <==
#
# Ansible managed
#
input(
  type="imfile"
  file="/var/log/php_errors.log"
  tag="php:"
  severity="err"
)
if
  ($syslogtag == "php:")
  then {
    call forwardToLogServer
}

==> 30-output-forwards-forwardToLogServer.conf <==
#
# Ansible managed
#
ruleset(name="forwardToLogServer") {
    *.* action(name="forwardToLogServer"
        type="omfwd"
        Target="psyslog.example.com"
        Port="514"
        Protocol="udp"
        Template="RSYSLOG_ForwardFormat"
    )
}
```

### CODE REVIEW

File: roles/rsyslog/vars/outputs/files/main.yml [hunk 1]
```diff
@@ -19,11 +19,15 @@ __rsyslog_conf_files_output_modules:
     type: 'modules'
     sections:
 
+      # yamllint disable rule:line-length
       - comment: 'Log messages into files with timestamp format either
```
Comment (richm) [inline roles/rsyslog/vars/outputs/files/main.yml:23]: does the choice of `logging_files_template_format` affect only the timestamp format, or does it affect other fields as well?

File: tests/tests_combination.yml [hunk 2]
```diff
@@ -145,13 +146,17 @@
...
-        /bin/grep
-        ' *input(type="imfile"
-        file="/var/log/inputdirectory/\*.log"
-        tag="{{ __test_tag }}"'
-        {{ __test_inputfiles_conf }}
-      changed_when: false
+      lineinfile:
+        path: "{{ __test_inputfiles_conf }}"
+        line: "  {{ item }}"
+        state: present
```
Comment (richm) [inline tests/tests_combination.yml:152]: nice - didn't know you could do this with lineinfile

File: roles/rsyslog/vars/outputs/files/main.yml [hunk 3]
```diff
@@ -19,11 +19,15 @@ __rsyslog_conf_files_output_modules:
...
-        traditional or default'
+        traditional, syslog or default'
         options: |-
-          {% if rsyslog_basics_use_traditional_timestamp_format | d(true) %}
```
Comment (richm) [inline roles/rsyslog/vars/outputs/files/main.yml:25]: is `rsyslog_basics_use_traditional_timestamp_format` still used?

Comment (nhosoi) [reply roles/rsyslog/vars/outputs/files/main.yml:25]: No, not any more. I decided to use the newly introduced param `logging_files_template_format` for the purpose instead. (Please note that `rsyslog_basics_use_traditional_timestamp_format` was not in the doc. so, I'd think it had no chance to be set...)

File: roles/rsyslog/vars/outputs/files/main.yml (hunk 1 above)
Comment (nhosoi) [reply roles/rsyslog/vars/outputs/files/main.yml:23]: Based on the doc https://www.rsyslog.com/doc/v8-stable/configuration/templates.html, it looks `RSYSLOG_SyslogProtocol23Format` affects other fields.
```
template(name="RSYSLOG_SyslogProtocol23Format" type="string"
     string="<%PRI%>1 %TIMESTAMP:::date-rfc3339% %HOSTNAME% %APP-NAME% %PROCID% %MSGID% %STRUCTURED-DATA% %msg%\n")
```

Comment (richm) [reply roles/rsyslog/vars/outputs/files/main.yml:23]: ok - then I guess change the comment to something like "Log messages into files with traditional, syslog, or default format" - or something like that

Comment (nhosoi) [reply roles/rsyslog/vars/outputs/files/main.yml]: Thank you, @richm. Fixed.

**Review by richm — APPROVED** [review]
lgtm

**Comment by richm** [comment]
unfortunately CI is down due to an outage, so it may be a while before we get test results :P

**Comment by nhosoi** [comment]
[citest bad]

**Comment by nhosoi** [comment]
Thanks, @richm. Merged.

**Pull request merged**